### ISSUE TITLE
 Cyrillic video titles loose their first word on conversion

### ISSUE THREAD

**Issue by djradivip** [Rudloff/alltube#272]
## Your environment

`ffmpeg version 3.4.7 Copyright (c) 2000-2019 the FFmpeg developers
built with gcc 4.8.5 (GCC) 20150623 (Red Hat 4.8.5-39)
configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --docdir=/usr/share/doc/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic' --extra-ldflags='-Wl,-z,relro ' --extra-cflags=' ' --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-version3 --enable-bzlib --disable-crystalhd --enable-fontconfig --enable-gcrypt --enable-gnutls --enable-ladspa --enable-libass --enable-libbluray --enable-libcdio --enable-libdrm --enable-indev=jack --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libmp3lame --enable-nvenc --enable-openal --enable-opencl --enable-opengl --enable-libopenjpeg --enable-libopus --disable-encoder=libopus --enable-libpulse --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libvidstab --enable-libx264 --enable-libx265 --enable-libxvid --enable-libzvbi --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-libmfx --enable-runtime-cpudetect
libavutil      55. 78.100 / 55. 78.100
libavcodec     57.107.100 / 57.107.100
libavformat    57. 83.100 / 57. 83.100
libavdevice    57. 10.100 / 57. 10.100
libavfilter     6.107.100 /  6.107.100
libavresample   3.  7.  0 /  3.  7.  0
libswscale      4.  8.100 /  4.  8.100
libswresample   2.  9.100 /  2.  9.100
libpostproc    54.  7.100 / 54.  7.100
`

**What is your operating system (Windows, Linux, OSX, etc.)?** CentOS 7

**What is your web server (Apache, IIS, etc.)?** OpenLiteSpeed

**What version of AllTube are you using?** 2.2.1

**How did you install AllTube (with Git or with a release package)?** Release, but I got ffmpeg and youtube-dl manually

**What version of PHP are you using?** 7.3

**What version of Python are you using?** 2.7.5

**What version of youtube-dl are you using?**  2020.03.24

**Do you get any PHP-related errors in your webserver's logs?** No

**What is the content of your `config/config.yml` file?**

```yaml
# Path to your youtube-dl binary
youtubedl: /usr/local/bin/youtube-dl

# Path to your python binary
python: /usr/bin/python

# An array of parameters to pass to youtube-dl
params:
    - --no-warnings
    - --ignore-errors
    - --flat-playlist

# True to enable audio conversion
convert: true

# True to enable advanced conversion mode
convertAdvanced: true

# List of formats available in advanced conversion mode
convertAdvancedFormats: [mp3, avi, flv, wav]

# Path to your avconv or ffmpeg binary
avconv: /usr/bin/ffmpeg

# avconv/ffmpeg logging level.
avconvVerbosity: error

# Path to the directory that contains the phantomjs binary.
phantomjsDir: vendor/bin/

# True to disable URL rewriting
uglyUrls: false

# True to stream videos through server
# Set to "ask" if you want to allow it but not enable by default.
stream: ask

# True to enable remux mode (merge best audio and best video)
remux: false

# MP3 bitrate when converting (in kbit/s)
audioBitrate: 128

# App name
appName: Time-Taker YouTube Downloader

# Generic formats supported by youtube-dl
genericFormats:
    best: Best
    bestaudio: Best Audio
    bestvideo+bestaudio: Remux best video with best audio
    worst: Worst

# Enable debug mode.
debug: false
```

**Please provide the URL of a video that causes the issue.**: https://www.youtube.com/watch?v=TuZSnWLS7f0

## Describe your issue
The problem consists in the fact that when I am downloading MP3 of a video, the title looses its first word in the name. This only happens with Cyrillic-titled videos and on the website only.

The problem isn't present when I download the video with " Stream the video through the server". 

However, as expected this downloads a video file. I tried having both checkboxes on my instance enabled(stream video and convert to MP3), but didn't fix it

On the server itself, the video's audio downloads fine but in .m4a format.

`
[root@static ~]# youtube-dl -f 'bestaudio' https://www.youtube.com/watch?v=TuZSnWLS7f0
[youtube] TuZSnWLS7f0: Downloading webpage
[download] Destination: Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.m4a
[download] 100% of 1.59MiB in 00:00
[ffmpeg] Correcting container in "Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.m4a"
`
I tried to convert it as well:

`
[root@static ~]# youtube-dl --extract-audio --audio-format mp3 https://www.youtube.com/watch?v=TuZSnWLS7f0
[youtube] TuZSnWLS7f0: Downloading webpage
[download] Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.m4a has already been downloaded
[download] 100% of 1.57MiB
[ffmpeg] Correcting container in "Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.m4a"
[ffmpeg] Destination: Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.mp3
Deleting original file Море пиле - Merry, Ellie & Lussy-TuZSnWLS7f0.m4a (pass -k to keep)
`
Conversion happened flawlessly and name got saved properly.

The issue doesn't happen with videos, where the title is made up only of latin letters.
As per my previous issue, I have "--restrict-filenames" removed from the config file, and I have the env variable, which you suggested in index.php.

Any ideas?

**Comment by Rudloff** [comment]
Thanks for the report.
The issue here is that we use `pathinfo()` to parse the filename and it needs an UTF-8 PHP locale to work correctly with Unicode strings.
I will try to see if we can use something else instead.

**Comment by djradivip** [comment]
What if we add this to index.php(under the line which sets environment)? Would you recommend such solution?

setlocale(LC_ALL, 'en_US.UTF-8');

**Comment by Rudloff** [comment]
This should now be fixed on the develop branch: a478aea

**Comment by djradivip** [comment]
Applied manually to my release. Thank you. :)
